google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/Exceptions.java

```diff
@@ -217,15 +217,20 @@ public static StorageException toStorageException(Throwable exception) {
 
   /**
    * This exception is thrown from {@link JsonStreamWriter#append()} when the client side Json to
-   * Proto serializtion fails. The exception contains a Map of indexes of faulty lines and the
-   * corresponding error message.
+   * Proto serializtion fails. It can also be thrown by the server in case rows contains invalid
+   * data. The exception contains a Map of indexes of faulty rows and the corresponding error
+   * message.
    */
-  public static class AppendSerializtionError extends RuntimeException {
+  public static class AppendSerializtionError extends StatusRuntimeException {
     private final Map<Integer, String> rowIndexToErrorMessage;
     private final String streamName;
 
-    public AppendSerializtionError(String streamName, Map<Integer, String> rowIndexToErrorMessage) {
-      super(String.format("Append serialization failed for writer: %s", streamName));
+    public AppendSerializtionError(
+        int codeValue,
```

Comment:
why not directly accept a Status? So for the response callback case, you don't need to disassemble it and assemble it.

Reply:
nvm